Sources/DeepLearning/Optimizer.swift

```diff
@@ -45,8 +45,9 @@ fileprivate extension Tensor where Scalar: Numeric {
 ///
 /// Reference: ["Adam - A Method for Stochastic Optimization"](
 /// https://arxiv.org/abs/1412.6980v8)
-public class Adam<Model: Layer>: Optimizer
-    where Model.AllDifferentiableVariables == Model.CotangentVector {
+public class Adam<Model: Differentiable & KeyPathIterable>: Optimizer
```

Comment:
This still has a chance to break existing notebooks are we are preparing with fast.ai.  I'd suggest not introducing any functional changes to existing abstractions until we are ready to do so.

Reply:
Oh I see. Ok I reverted that.

Reply:
Yeah, we are in a pre-rerelease phase and the fast.ai course is next week. Thanks for understanding! Let's revisit this in a few weeks.